src/content/developers/docs/scaling/zk-rollups/index.md

```diff
@@ -1,34 +1,206 @@
+
+The ZK-rollup smart contract maintains the state of all transactions on the L2 chain, and this state can only be updated with a validity proof. This means that ZK-rollups only need to provide validity proofs to finalize transactions on the layer 1 (L1) chain instead of posting all the transaction data like [optimistic rollups](/developers/docs/scaling/optimistic-rollups/). 
+
+There are no delays when moving funds from ZK-rollups (L2) to Ethereum (L1) because exit transactions are finalized once the ZK-rollup contract verifies the validity proof. Conversely, withdrawing funds from optimistic rollups is subject to a delay to allow anyone to challenge the exit transaction with a [fraud proof](/glossary/#fraud-proof). 
+
+ZK-rollups write transactions to Ethereum as `calldata`, which allows anyone to reconstruct the rollup’s state independently. They also use compression techniques to reduce transaction data—for example, accounts are represented by an index rather than an address, which saves 28 bytes of data. On-chain data publication is a significant cost for rollups, so data compression can reduce fees for users.  
+
+## How do zk-rollups interact with Ethereum? {#zk-rollups-and-ethereum}
+
+A ZK-rollup is an L2 chain designed to operate on top of Ethereum’s base layer. The ZK-rollup protocol is managed by smart contracts running on L1, including the main contract that stores rollup blocks, tracks deposits, and monitors state updates. Another L1 contract (the verifier contract) verifies zero-knowledge proofs submitted by block producers. 
+
+Although they operate independently, ZK-rollups directly inherit Ethereum's security guarantees. This makes them safer than scaling solutions, such as [sidechains](/developers/docs/scaling/sidechains/), which are responsible for their own consensus and security properties.  
+
+ZK-rollups rely on the main Ethereum protocol for the following:  
+
+### Data availability {data-availability}
+
+ZK-rollups publish state data for every transaction processed off-chain to Ethereum. With this data, it is possible for individuals or businesses to reproduce the rollup’s state and validate the chain themselves. Ethereum makes this data available to all participants of the network as `calldata`. 
+
+ZK-rollups don’t actually need to publish a lot of transaction data on-chain since validity proofs already verify the authenticity of state transitions. Nonetheless, storing data on-chain is still important for many reasons. Putting state data on-chain allows anyone to rebuild the L2 chain's state and produce batches, preventing malicious operators from arbitrarily freezing the chain. 
```

Comment:
```suggestion
ZK-rollups don’t need to publish much transaction data on-chain because validity proofs already verify the authenticity of state transitions. Nevertheless, storing data on-chain is still important because it allows permissionless, independent verification of the L2 chain's state which in turn allows anyone to submit batches of transactions, preventing malicious operators from censoring or freezing the chain. 
```